src/osd/strconv.h

```diff
@@ -25,22 +25,30 @@
 
 #include <windows.h>
 
-// the result of these functions has to be released with osd_free()
-
+std::string astring_from_utf8(const std::string &s);
 std::string astring_from_utf8(const char *s);
+std::string &astring_from_utf8(std::string &dst, std::string &s);
```

Comment:
'astring' in this context has absolutely nothing to do with MAME's old 'astring'.  In this context, 'astring' is an "ANSI Code Page" string, using the so-called Win32 ANSI code page.

There is an argument to be made in 2016, supporting this is pointless, but we should arguably dismantle everything TCHAR-related if that is the case.